api/oneclick.py

```diff
@@ -688,7 +734,11 @@ def validate_patron(self, patron, create=False):
         if not patron_cardno:
             raise InvalidInputException("Patron %r has no card number.", patron)
 
-        patron_oneclick_id = self.get_patron_internal_id(patron_cardno=patron_cardno)
+        try:
+            patron_oneclick_id = self.get_patron_internal_id(patron_cardno=patron_cardno)
+        except NotFoundOnRemote, e:
+            patron_oneclick_id = None
+            
```

Comment:
This was raising an exception when the patron had no OneClick account, and it was halting the request even when `create=True`, which was supposed to detect this situation and fix it.